lib/decorators/component-list.js

```diff
@@ -163,12 +163,16 @@ function addComponentList(el, componentList) {
  * decorate component list if specified in the schema
  * @param  {string} uri
  * @param  {Element} el with data-editable
+ * @returns {Promise|null}
  */
 export default function handler(uri, el) {
   const path = el.getAttribute(editAttr),
     schema = getSchema(uri, path); // note: component lists are ALWAYS a single field, so we don't need co call groups.get
 
   if (schema && schema[componentListProp]) {
     addComponentList(el, schema[componentListProp]);
+    if (getData(uri, path) === undefined) {
+      return store.dispatch('saveComponent', { uri, data: { [path]: [] }});
```

Comment:
Hmm, the stuff that calls this decorator doesn't expect it to return a promise, but this is a valid use case. Let me take a look today / tomorrow morning and update this to cover all edge cases (e.g. if the list is null, or some other non-array value).